packages/cli/README-source.md

```diff
@@ -724,6 +724,49 @@ The `z.cursor` object exposes two methods:
 
 Any data you `set` will be available to that Zap for about an hour (or until it's overwritten). For more information, see: [paging](#paging).
 
+### `z.generateCallbackUrl()`
+
+The `z.generateCallbackUrl()` will return a callback URL your app can `POST` to later for handling long running tasks (like transcription or encoding jobs). In the meantime, the Zap and Task will wait for your response and the user will see the Task marked as waiting.
+
+For example, in your `perform` you might do:
+
+```js
+const perform = async (z, bundle) => {
+  // something like this url:
+  // https://zapier.com/hooks/callback/123/deadbeef-dead-dead-dead-deaddeafbeef/abcdef0123456789abcdef0123456789abcdef01/
+  const callbackUrl = z.generateCallbackUrl();
+  await z.request({
+    url: 'https://example.com/api/slow-job',
+    method: 'POST',
+    body: {
+      // ... whatever your app needs
+      url: callbackUrl,
+    },
+  });
+  return {"hello": "world"};
+};
+```
+
+And in your own `/api/slow-job` view (or more likely, an async job) you'd make this request to Zapier:
+
+```http
+POST /hooks/callback/123/deadbeef-dead-dead-dead-deaddeafbeef/abcdef0123456789abcdef0123456789abcdef01/ HTTP/1.1
+Host: zapier.com
+Content-Type: application/json
+
+{"foo":"bar"}
+```
+
+And finally, in a `performResume` to handle the final step (which will receive `bundle.cleanedRequest`, `bundle.rawRequest`, and `bundle.outputData`):
```

Comment:
I'd clarify which `bundle.X` properties come from where. It looks like

* `bundle.cleanedRequest` is `{foo: "bar"}`, the payload for the callback url
* `bundle.rawRequest` is the request object for `bundle.cleanedRequest`
* `bundle.outputData` is the data returned from the initial `perform`